KTL.js

```diff
@@ -5620,6 +5622,145 @@ function Ktl($, appInfo) {
+                        resolve();
+                    });
+                });
+
+                const callStickTableHeadersIfNotResolved = () => {
+                    if (!isCalledInsideObserver) {
+                        callStickTableHeaders();
+                    }
+                };
+
+                // If observerPromise does not resolve within 1000ms, call callStickTableHeadersIfNotResolved
+                setTimeout(callStickTableHeadersIfNotResolved, 1000);
+            } else {
+                callStickTableHeaders();
+            }
+        }
+
+        /** _stc = Sticky Table Columns
+        * keyword @param stickyColBkgdColor - The background color of the sticky column
+        * @param {string} viewId - The view.key
+        * @param {object} keywords - The keywords object*/
+        function stickyTableColumns(viewId, keywords) {
+            const kw = '_stc';
+            let stickyColBkgdColor = 'rgb(243 246 249)';
+            let numOfColumns = 1;
+            let isCalledInsideObserver = false;
+
+            if (!keywords[kw]) return;
+
+            if (keywords[kw].length) {
+                numOfColumns = keywords[kw][0].params[0][0] || numOfColumns;
+                stickyColBkgdColor = keywords[kw][0].params[0][1] || stickyColBkgdColor;
+            }
+
+            const callStickTableColumns = () => {
+                if (!isCalledInsideObserver) {
+                    ktl.views.stickTableColumns(viewId, numOfColumns, stickyColBkgdColor);
+                }
+            }
+
+            if (isBulkOpsEnabled(keywords, viewId)) {
+                numOfColumns = Math.max(numOfColumns, 2);
+                let observerPromise = new Promise((resolve) => {
+                    observeAndExecute(`#bulkOpsControlsDiv-${viewId}`, () => {
+                        ktl.views.stickTableColumns(viewId, numOfColumns, stickyColBkgdColor);
+                        isCalledInsideObserver = true;
+                        resolve();
+                    });
+                });
+
+                const callStickTableColumnsIfNotResolved = () => {
+                    if (!isCalledInsideObserver) {
+                        callStickTableColumns();
+                    }
+                };
+
+                // If observerPromise does not resolve within 1000ms, call callStickTableColumns
+                setTimeout(callStickTableColumnsIfNotResolved, 1000);
+            } else {
+                callStickTableColumns();
+            }
+        }
+
+        function isBulkOpsEnabled(keywords, viewId) {
+            if (keywords._nbo) return false;
+
+            const bulkOps = [
+                { operation: 'bulkEdit', role: 'Bulk Edit', class: 'cell-editable' },
+                { operation: 'bulkCopy', role: 'Bulk Copy', class: 'cell-editable' },
+                { operation: 'bulkDelete', role: 'Bulk Delete', selector: 'kn-delete-link' },
+            ];
+
+            return bulkOps.some(op => {
```

Comment:
You could leverage the function `filter` before `some`. Code like `if (!isEnabled) return false;` will be easier to read.